### ISSUE TITLE
Query exists-match feature

### ISSUE THREAD

**Issue by wkordalski** [tree-sitter/tree-sitter#1703]
I have code like this:

```python
def f():
	a + b
```

Which parses to:

```
module [0, 0] - [2, 0]
  function_definition [0, 0] - [1, 6]
    name: identifier [0, 4] - [0, 5]               (*)
    parameters: parameters [0, 5] - [0, 7]
    body: block [1, 1] - [1, 6]
      expression_statement [1, 1] - [1, 6]
        binary_operator [1, 1] - [1, 6]
          left: identifier [1, 1] - [1, 2]
          right: identifier [1, 5] - [1, 6]
```

I also have a query:

```
(function_definition name: _ @fnname)
```

And I have a node — identifier `f` — marked in the tree with `(*)`.

I would like to know if there exists a match of the query where `@fnname` == "the node marked with `(*)`".

Is it possible in Tree-sitter?

----

I can iterate over all matches, but this is not optimal.

**Comment by ris58h** [comment]
Do you want to match nodes where `@fnname == "f"`?
If so, you should use `eq` predicate (see `Predicates` section in [Query Syntax](https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax)):
```
(
  (function_definition name: (identifier) @fnname)
  (#eq? @fnname "f")
)
```

**Comment by wkordalski** [comment]
No. I want to know if there exists a match on given tree where `@fnname` is some given node in the tree.

**Comment by amaanq** [comment]
Sorry for responding so late, but I'm kind of confused as to what you're asking, do you just want functionality where, given a list of matches, you can figure out if a given capture exists in the matches? I don't really understand the point of that...

I'll close this for now, but if this is still an issue for you and you can elaborate more on what exactly you're looking for, that'd be really helpful
